engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -125,6 +126,8 @@
 	 */
 	private final List<ValidatedValueUnwrapper<?>> validatedValueHandlers;
 
+	private final ClassLoader userClassLoader;
```

Comment:
Btw, do you think that `userClassLoader` is the most fitting name? Somehow I always think of 'end user' in this case. What's about `providedClassLoader`. It's maybe a bit more neutral.